### DISCUSSION TITLE
auroraのアップデート

### DISCUSSION THREAD

**Discussion by RYOTASUDA** [kakeru-one/web-engineering-roundtable#76] [General]
test

**Comment by RYOTASUDA** [comment]
## 前提
- プライマリークラスター（aurora2, mysql5.7）（クラスター、インスタンス）
  - クラスターパラメーターグループ（5.7）
  - パラメーターグループ（5.7）

**Comment by RYOTASUDA** [comment]
![mysql-blue-green-upgrade](https://github.com/user-attachments/assets/31d268bf-b4fb-4b46-9da8-75ddf421c44d)

**Comment by RYOTASUDA** [comment]
## 事前準備手順
1. 旧クラスターと全く同じクラスターを作る
  - 参照するパラメーターグループ、クラスターパラメーターグループは同じ

2. 旧パラメーターグループのコピーを作成する
3. コピーしたパラメーターグループの設定値をaurora3で使えるものにする
4. 新たに作成したクラスターのaurora, mysqlのバージョンを上げつつ、参照するパラメータグループを変更する

**Comment by RYOTASUDA** [comment]
## 当日手順
1. 新たに作成したクラスターをプライマリーに昇格させる
2. CNAMEレコードのvalueを旧クラスターから新クラスターに変更する

**Reply by yuyakawai0411** [reply]
> 新たに作成したクラスターをプライマリーに昇格させる

もしmysql 8.0側でトラブルが起きたとしても、プライマリーで起きた変更がレプリカに同期されるため、mysql 5.7側に戻すときに特別な手順を踏まなくてよい
